### PR TITLE
Resolve config directory relative to EmberApp project root

### PR DESCRIPTION
**Pull request by nickiaconis**
Similar in concept to #5614, but for the config directory.

### CODE REVIEW

**Comment by stefanpenner** [comment]
:+1:

**Comment by chadhietala** [comment]
Windows has opinions about paths.

**Comment by nickiaconis** [comment]
@chadhietala I think you're referring to the "Starts with /" part, yes? I could do "Starts with project.root" instead. Thoughts?

File: tests/unit/models/project-test.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ describe('models/project.js', function() {
 
       var expected = path.normalize('tests/dummy/config/environment');
 
-      expect(project.configPath()).to.equal(expected);
+      expect(project.configPath()).to.match(new RegExp(expected + '$'));
```
Comment (Turbo87) [inline tests/unit/models/project-test.js]: can we use `expect(...).to.contain()` here?

**Comment by chadhietala** [comment]
@nickiaconis I think you need to account for `\`. The errors in Appveyor seem to be issues with finding files.

File: tests/unit/models/project-test.js (hunk 1 above)
Comment (nickiaconis) [inline tests/unit/models/project-test.js]: We could. I thought the more specific assertion, that the expected path exists at the very end of the output, was a better check than the more generic one, that the expected path exists somewhere within the output.

**Comment by nickiaconis** [comment]
@chadhietala Yes, it looks like it was falling back to the `else` case on appveyor. I changed the condition for the middle case, and I think it should work for both Unix and non-Unix paths now.

**Comment by nickiaconis** [comment]
Okay, I think I'm going about this wrong. `lib/model/project` worked fine as it was, so I'll undo those changes. This only affects custom config paths.

What is the expectation for custom config paths? If my config file lives at `project_root/ember-config/environment.js` what value should `emberAddon.configPath` take in `package.json`? `'ember-config'` or `'ember-config/environment'`? (My guess is the latter.)

**Comment by nathanhammond** [comment]
It should take the latter, `ember-config/environment`, based upon the existing code paths and tests.

**Comment by nickiaconis** [comment]
Okay, for my future self's benefit and the benefit of others, [the origin of the issue lies in the `EmberApp#_configTree` method](https://github.com/ember-cli/ember-cli/blob/master/lib/broccoli/ember-app.js#L970). That line assumes that `Project#configPath` returns a fully resolved path. Therefore, `Project#configPath` must be prepended `Project#root` to the relative configuration path. This ensures uses of `Project#configPath` from methods other than `Project#config` work as expected.

**Comment by nathanhammond** [comment]
Okay, so we might have this resolved. Kicking over AppVeyor.

**Comment by nickiaconis** [comment]
@nathanhammond Has this been unblocked by #5774?

**Comment by nickiaconis** [comment]
@nathanhammond How did I not see your message? /)_-

I think it needed a rebase to resolve the issue. I've done that, which kicked off a new build.

**Comment by nathanhammond** [comment]
[It should eventually pass assuming your code works.](https://ci.appveyor.com/project/embercli/ember-cli/build/7821) Kicked it over a second ago for what looked like a transient issue.

**Comment by nickiaconis** [comment]
Passed 4.2, but failed 5.0. I can't tell if it's a false-negative though; it doesn't look like the false-negatives I've seen in the past. @nathanhammond can you take a look?

**Comment by nathanhammond** [comment]
I've not seen that before, seems like it might be a problem from your commit? Kicking it over again anyway.

**Comment by nathanhammond** [comment]
@homu r+

**Pull request merged**